pkg/dao/repository.go

```diff
@@ -367,13 +367,17 @@ func (r repositoryDaoImpl) ValidateParameters(orgId string, params api.Repositor
 	return response, err
 }
 
-func (r repositoryDaoImpl) ValidateName(orgId string, name string, response *api.GenericAttributeValidationResponse) error {
+func (r repositoryDaoImpl) ValidateName(orgId string, name string, response *api.GenericAttributeValidationResponse, uuids []string) error {
 	if name == "" {
 		response.Valid = false
 		response.Error = "Name cannot be blank"
 	} else {
 		found := models.RepositoryConfiguration{}
-		result := r.db.Where("name = ? AND ORG_ID = ?", name, orgId).Find(&found)
+		query := r.db.Where("name = ? AND ORG_ID = ?", name, orgId)
+		if len(uuids) != 0 {
+			query = query.Where("repository_configurations.uuid NOT IN ?", uuids)
+		}
+		result := query.Find(&found)
```

Comment:
I'm having trouble understanding what this PR is for just from reading the description on the issue. Could you explain what the purpose of this change is? It looks like this query is the meat of the change.

Reply:
Correct! I will add description... 💯